### PR TITLE
Add virus total certified

### PR DESCRIPTION
**Pull request by shschwar**
Add virus total certified

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by shschwar** [comment]
Thanks @Doug Elliott (Artech Consulting LLC)<mailto:v-wiell@microsoft.com>

@Yehuda Tognder (Matchpoint IT Ltd)<mailto:v-yetogn@microsoft.com> – can you please provide a readme file and I will update the repository?
Thanks

File: certified-connectors/VirusTotal/apiProperties.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "properties": {
+    "connectionParameters": {
+      "api_key": {
+        "type": "securestring",
+        "uiDefinition": {
+          "displayName": "x-api_key",
+          "description": "The x-api_key for this api",
+          "tooltip": "Provide your x-api_key",
+          "constraints": {
+            "tabIndex": 2,
```
Comment (ShyamSunder90) [inline certified-connectors/VirusTotal/apiProperties.json:11]: It starts with 1

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,3325 @@
...
+            "schema": {
+              "$ref": "#/definitions/filesReport"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "analysesResponses": {
```
Comment (ShyamSunder90) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:157]: This is huge... Do you need all properties or can it be resricted to few important/most used properties? If yes, how are you expecting uses to use it? This looks like a usability issue

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,3325 @@
...
+              "type": "string"
+            },
+            "type": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "ipResult": {
```
Comment (ShyamSunder90) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:2043]: This is huge... Do you need all properties or can it be resricted to few important/most used properties? If yes, how are you expecting uses to use it? This looks like a usability issue

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,3325 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.4",
```
Comment (ShyamSunder90) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:4]: Just wanted to double check.. did you run swagger validator on this to make sure there are no issues?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,3325 @@
...
+    "title": "Virus Total",
+    "description": "Virus Total is an online service that analyzes suspicious files and URLs to detect types of malware and malicious content using antivirus engines and website scanners. It provides an API that allows users to access the information generated by VirusTotal.",
+    "x-ms-api-annotation": {
+      "status": "Preview"
```
[the remark this answers is not in the document]
Comment (shschwar) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:8]: remove what? "      
"status": "Preview"
"?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 2 above)
Comment (shschwar) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:157]: Yes, we have review the requirement with the PM, and discuss the large amount of properties, After double verify it have come to understanding that it must have all the properties

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 3 above)
Comment (shschwar) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:2043]: Yes, we have review the requirement with the PM, and discuss the large amount of properties, After double verify it have come to understanding that it must have all the properties

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 4 above)
Comment (shschwar) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:4]: It pass all the validations and certifications so I guess it does

File: certified-connectors/VirusTotal/apiProperties.json (hunk 1 above)
Comment (shschwar) [reply certified-connectors/VirusTotal/apiProperties.json]: fixed to 1

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (ShyamSunder90) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json]: x-ms-api-annotation enitre object

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 2 above)
Comment (ShyamSunder90) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:154]: Can you please connect your PM with me. I am concerned about the user experience

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (shschwar) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json]: done

File: certified-connectors/VirusTotal/readme.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+ ### NOTE
+> This is a *sample* connector.  The connector is provided here with the intent to illustrate certain features and functionality around connectors.
```
Comment (dougelliott) [inline certified-connectors/VirusTotal/readme.md:2]: This is a certified connector, not a sample one. So this whole note should be removed.

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,3443 @@
...
+    "/api/v3/ip_addresses/{ip}": {
+      "get": {
+        "summary": "Ip scan report V3",
+        "description": "scan Ip to get full report",
+        "operationId": "VirusTotalGetIpScanV3",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "ip",
+            "description": "Ip to verify",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:31]: Can you change to `IP address` to match the description with the one in https://developers.virustotal.com/v3.0/reference#ip-info?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,3443 @@
...
+    "/api/v3/analyses/{id}": {
+      "get": {
+        "summary": "Retrieve information about a file or URL analysis",
+        "description": "Retrieve information about a file or URL analysis. This action uses the output id of Analyse an URL or Upload and analyse a file.",
+        "operationId": "VirusTotalRetrieveInfo",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "report",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:85]: Description should begin with upper case letter. Why `report` is more appropriate than `Analysis identifier` as in https://developers.virustotal.com/v3.0/reference#analysis?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,3443 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.4",
+    "title": "Virus Total APIV3",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:5]: Any reason to club `API` and `V3` into a single word? Can the title be changed to `VirusTotal API version 3` as in https://developers.virustotal.com/v3.0/reference#overview?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,3443 @@
...
+    "description": "user will be able to enter an ip address and get the exact experience of a report in virus total on this IP.",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:6]: - Start connector description with upper case letter.
- User upper case for acronyms like `IP`.
- Replace `virus total` with `VirusTotal`.
- What does `on this IP.` refer to?

File: certified-connectors/VirusTotal/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+
+Analyze suspicious files and URLs to detect types of malware, automatically share them with the security community.
+
+
+## Pre-requisites
+
+To use this integration, you need to have a VirusTotal account. Follow steps on https://www.virustotal.com/ to get your API Key.
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/readme.md:7]: - Replace `https://www.virustotal.com/` with `https://developers.virustotal.com/v3.0/reference#getting-started`.
- Replace `Key` with `key`.
- Remove space between `[]` and `()` in like 15 to make it a valid Markdown link.

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 7 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:31]: done

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 8 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:85]: Done

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 9 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:5]: Done

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 10 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:6]: Done

File: certified-connectors/VirusTotal/readme.md (hunk 11 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/readme.md:7]: done

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,3294 @@
+	{
+	  "swagger": "2.0",
+	  "info": {
+		"version": "1.0.4",
+		"title": "Virus Total",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:5]: It's a single word (See https://en.wikipedia.org/wiki/VirusTotal). Why are there spaces between everywhere between 'Virus' and 'Total'?

Comment (JatinSanghvi) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:5]: Not a quality pull request when we cannot spell the registered service name correctly 😕. What's the reason for using 'Ip' with lower-case 'p'? Such text descriptions might take the customers off from the power platform.

**Comment by sriyen-msft** [comment]
Followed up with the partner in certification portal to address the comments mentioned in the PR.

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 12 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json]: fix

**Comment by ytognder** [comment]
all review are fix with commit from  Jul 12, 2020

**Comment by mamurshe** [comment]
@JatinSanghvi @sriyen-msft Please take a look.

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,3294 @@
+	{
+	  "swagger": "2.0",
+	  "info": {
+		"version": "1.0.4",
+		"title": "VirusTotal",
+		"description": "Virus Total is an online service that analyzes suspicious files and URLs to detect types of malware and malicious content using antivirus engines and website scanners. It provides an API that allows users to access the information generated by VirusTotal.",
```
Comment (JatinSanghvi) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:6]: `Virus Total` to `VirusTotal` ?

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,3294 @@
...
+		"/api/v3/analyses/{id}": {
+		  "get": {
+			"summary": "Retrieve information about a file or URL analysis",
+			"description": "Retrieve information about a file or URL analysis. This action uses the output id of Analyse an URL or Upload and analyse a file.",
+			"operationId": "VirusTotalRetrieveInfo",
+			"parameters": [
+			  {
+				"in": "path",
+				"name": "id",
+				"description": "Retrieve information about a file or URL analysis",
```
Comment (sriyen-msft) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:89]: @ytognder The apiDefinition.swagger.json file is outdated - Please update the latest one you have uploaded in the certification portal

"description": "Retrieve information about a file or URL analysis" -> should be "description": "report",

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 15]
```diff
@@ -0,0 +1,3294 @@
...
+	  "consumes": [],
+	  "produces": [
+		"application/json"
+	  ],
+	  "paths": {
+		"/api/v3/ip_addresses/{ip}": {
+		  "get": {
+			"summary": "IP scan report",
+			"description": "Retrieve information about an IP address",
+			"operationId": "VirusTotalGetIpScan",
```
Comment (sriyen-msft) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:30]: @ytognder The apiDefinition.swagger.json file is outdated - Please update the latest one you have uploaded in the certification portal

 "/api/v3/ip_addresses/{ip}": {
            "get": {
                "description": "Retrieve information about an IP address",
                "operationId": "VirusTotalGetIpScan",
                "parameters": [
                    {
                        "description": "IP to verify",
                        "in": "path",
                        "name": "ip",
                        "required": true,
                        "type": "string",
                        "x-ms-summary": "IP Address",
                        "x-ms-url-encoding": "single"
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Ok",
                        "schema": {
                            "$ref": "#/definitions/ipResult"
                        }
                    }
                },
                "summary": "IP scan report"
            }
        },

should be
"/api/v3/ip_addresses/{ip}": {
            "get": {
                "description": "Scan IP address to get full report",
                "operationId": "VirusTotalGetIpScanV3",
                "parameters": [
                    {
                        "description": "Ip to verify",
                        "in": "path",
                        "name": "ip",
                        "required": true,
                        "type": "string",
                        "x-ms-summary": "IP Address",
                        "x-ms-url-encoding": "single"
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Ok",
                        "schema": {
                            "$ref": "#/definitions/ipResult"
                        }
                    }
                },
                "summary": "Ip scan report V3"
            }
        },

File: certified-connectors/VirusTotal/apiDefinition.swagger.json [hunk 16]
```diff
@@ -0,0 +1,3294 @@
...
+		  }
+		},
+		"/api/v3/urls": {
+		  "post": {
+			"consumes": [
+			  "multipart/form-data"
+			],
+			"summary": "Analyse an URL",
+			"description": "Analyse an URL",
+			"operationId": "VirusTotalAnalysisurl",
```
Comment (sriyen-msft) [inline certified-connectors/VirusTotal/apiDefinition.swagger.json:59]: @ytognder The apiDefinition.swagger.json file is outdated - Please update the latest one you have uploaded in the certification portal..

"operationId": "VirusTotalAnalysisurl" should be "operationId": "VirusTotalAnalysisurlV3"

File: certified-connectors/VirusTotal/apiProperties.json [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+  "properties": {
+    "connectionParameters": {
+      "api_key": {
+        "type": "securestring",
+        "uiDefinition": {
+          "displayName": "x-api_key",
+          "description": "The x-api_key for this api",
+          "tooltip": "Provide your x-api_key",
+          "constraints": {
+            "tabIndex": 1,
```
Comment (sriyen-msft) [inline certified-connectors/VirusTotal/apiProperties.json:11]: @ytognder Please change it to 2 as per the latest version of your connector.

"tabIndex": 1 -> "tabIndex": 2

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 14 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:89]: image

will not fix see attach image from Virus Total web site

File: certified-connectors/VirusTotal/apiDefinition.swagger.json (hunk 16 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiDefinition.swagger.json:59]: will not change the is the right name

File: certified-connectors/VirusTotal/apiProperties.json (hunk 17 above)
Comment (ytognder) [reply certified-connectors/VirusTotal/apiProperties.json:11]: image

@ShyamSunder90  request to change it to 1  see attach image.

**Pull request merged**